### DISCUSSION TITLE
Nested fields (Add related resources at creation time)

### DISCUSSION THREAD

**Discussion by adrianthedev** [avo-hq/avo#779] [Feature requests]
Based on a few [conversations](https://discord.com/channels/740892036978442260/740893011994738751/959950886292058122) we had with users, having a  way to add related records at creation time is something that would benefit Avo users.

### Related resources
 - https://github.com/nathanvda/cocoon
 - https://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html
 - https://www.pluralsight.com/guides/ruby-on-rails-nested-attributes
 - https://medium.com/@e_j_gann/the-enchantment-of-rails-nested-attributes-5dd74d05d109

**Comment by bear-in-mind** [comment]
Hey @adrianthedev !
This is actually one of the reasons I had to monkey patch `avo` for my use. I'd love to help on that topic, especially since [I built a gem](https://github.com/eki-177/abyme) for this exact purpose :)

**Reply by adrianthedev** [reply]
That looks amazing! I'd love to have that kind of functionality in Avo. 💪

Let me know what I can do to help you and navigate the codebase.

**Comment by adrianthedev** [comment]
I posted [this](https://docs.avohq.io/2.0/recipes/nested-records-when-creating.html) guide that shows an initial iteration on how nested records on create screen can be right now.

**Comment by Paul-Bob** [comment]
https://docs.avohq.io/3.0/associations/has_many.html#nested-in-forms
